bugbot/rules/inactive_revision.py

```diff
@@ -0,0 +1,349 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+from typing import Dict, List
+
+from dateutil.relativedelta import relativedelta
+from jinja2 import Environment, FileSystemLoader, Template
+from libmozdata import utils as lmdutils
+from libmozdata.connection import Connection
+from libmozdata.phabricator import PhabricatorAPI
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from bugbot import utils
+from bugbot.bzcleaner import BzCleaner
+from bugbot.history import History
+from bugbot.user_activity import PHAB_CHUNK_SIZE, UserActivity
+
+PHAB_FILE_NAME_PAT = re.compile(r"phabricator-D([0-9]+)-url\.txt")
+PHAB_TABLE_PAT = re.compile(r"^\|\ \[D([0-9]+)\]\(h", flags=re.M)
+
+
+class InactiveRevision(BzCleaner):
+    """Bugs with inactive patches that are awaiting action from authors or reviewers."""
+
+    def __init__(self, old_patch_months: int = 6, patch_activity_months: int = 6):
+        """Constructor
+
+        Args:
+            old_patch_months: number of months since creation of the patch to be
+                considered old. If the bug has an old patch, we will mention
+                abandon the patch as an option.
+            patch_activity_months: Number of months since the last activity on the patch.
+        """
+        super(InactiveRevision, self).__init__()
```

Comment:
All other rules in BugBot include the rule name and `self`, so I was thinking we could keep it as it is for consistency. Please let me know if you think otherwise and I can go ahead with this change.